### PR TITLE
test: migrate auth and group config validation tests

### PR DESCRIPTION
**Pull request by camposperez-hub**
**1. Description**
Migrates auth and group configuration parsing and validation unit tests from `internal/server/server_test.go` into their respective packages' test files (`internal/group/`, `internal/auth/google/`, and `internal/auth/generic/`).
- Implements `TestParseFromYaml()` and `TestFailParseFromYaml()` in each target package following the pattern in `alloydb_pg_test.go`.
- Removes `TestGroupConfigParsing()`, `TestGroupConfigValues()`, `TestGoogleAuthConfigValidation()`, and `TestGenericAuthConfigValidation()` from `server_test.go`.
- Adds `export_test.go` helpers in `google` and `generic` to prevent circular test dependencies with `internal/server`.

### CODE REVIEW

File: internal/group/group_test.go [hunk 1]
```diff
@@ -351,8 +506,8 @@ func TestFailParseFromYaml(t *testing.T) {
 				t.Fatalf("expect parsing to fail")
 			}
 			errStr := err.Error()
-			if !strings.Contains(errStr, tc.err) {
-				t.Fatalf("unexpected error: got %q, want it to contain %q", errStr, tc.err)
+			if errStr != tc.err {
+				t.Fatalf("unexpected error: got %q, want %q", errStr, tc.err)
 			}
```
[the remark this answers is not in the document]
Comment (camposperez-hub) [reply internal/group/group_test.go:511]: added in 00b18b85546d01003bc48e68265bd3e7322452b5

File: internal/group/group_test.go [hunk 2]
```diff
@@ -340,7 +437,65 @@ func TestFailParseFromYaml(t *testing.T) {
 			name: my-group
 			ttlMs: -100
 			`,
-			err: "Field validation for 'TTLMs' failed on the 'gte' tag",
+			err: "error unmarshaling group: unable to unmarshal group: [2:8] Key: 'GroupConfig.TTLMs' Error:Field validation for 'TTLMs' failed on the 'gte' tag\n   1 | name: my-group\n>  2 | ttlMs: -100\n              ^\n",
```
[the remark this answers is not in the document]
Comment (camposperez-hub) [reply internal/group/group_test.go]: added in 00b18b85546d01003bc48e68265bd3e7322452b5

File: internal/auth/generic/generic_test.go [hunk 3]
```diff
@@ -935,10 +936,139 @@ func TestIsJWTFormat(t *testing.T) {
...
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			_, _, _, _, _, _, err := server.UnmarshalPrimitiveConfig(context.Background(), testutils.FormatYaml(tc.in))
+			if err == nil {
+				t.Fatalf("expect parsing to fail")
+			}
+			errStr := err.Error()
+			if errStr != tc.err {
+				t.Fatalf("unexpected error: got %q, want %q", errStr, tc.err)
+			}
```
[the remark this answers is not in the document]
Comment (camposperez-hub) [reply internal/auth/generic/generic_test.go]: added in 00b18b85546d01003bc48e68265bd3e7322452b5

File: internal/group/group_test.go [hunk 4]
```diff
@@ -340,7 +437,65 @@ func TestFailParseFromYaml(t *testing.T) {
...
+		},
+		{
+			desc: "unknown field",
+			in: `
+			kind: group
+			name: my_group
+			resources:
+			  - res_a
+			`,
+			err: "error unmarshaling group: unable to unmarshal group: [2:1] unknown field \"resources\"\n   1 | name: my_group\n>  2 | resources:\n       ^\n   3 | - res_a",
```
[the remark this answers is not in the document]
Comment (camposperez-hub) [reply internal/group/group_test.go]: added in 00b18b85546d01003bc48e68265bd3e7322452b5

File: internal/auth/google/google_test.go [hunk 5]
```diff
@@ -231,3 +234,139 @@ func TestValidateMCPAuth_Opaque_Fallback(t *testing.T) {
...
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			_, _, _, _, _, _, err := server.UnmarshalPrimitiveConfig(context.Background(), testutils.FormatYaml(tc.in))
+			if err == nil {
+				t.Fatalf("expect parsing to fail")
+			}
+			errStr := err.Error()
+			if errStr != tc.err {
+				t.Fatalf("unexpected error: got %q, want %q", errStr, tc.err)
+			}
```
[the remark this answers is not in the document]
Comment (camposperez-hub) [reply internal/auth/google/google_test.go]: added in 00b18b85546d01003bc48e68265bd3e7322452b5

File: internal/group/group_test.go [hunk 6]
```diff
@@ -331,7 +428,7 @@ func TestFailParseFromYaml(t *testing.T) {
 			name: my-group
 			cacheScope: secret
 			`,
-			err: "Field validation for 'CacheScope' failed on the 'oneof' tag",
+			err: "error unmarshaling group: unable to unmarshal group: [1:13] Key: 'GroupConfig.CacheScope' Error:Field validation for 'CacheScope' failed on the 'oneof' tag\n>  1 | cacheScope: secret\n                   ^\n   2 | name: my-group",
```
[the remark this answers is not in the document]
Comment (camposperez-hub) [reply internal/group/group_test.go]: added in 00b18b85546d01003bc48e68265bd3e7322452b5

File: internal/auth/generic/export_test.go
Comment (Yuan325) [inline internal/auth/generic/export_test.go:1]: lets not open a new test

File: internal/auth/google/export_test.go
Comment (Yuan325) [inline internal/auth/google/export_test.go:1]: lets not add new tests~

**Review by Yuan325 — CHANGES_REQUESTED** [review]

File: internal/auth/generic/export_test.go
Comment (camposperez-hub) [reply internal/auth/generic/export_test.go:1]: export_test.go actually doesn't add any new tests—it contains zero test cases.

It's just a Go helper to avoid an import cycle: internal/server already imports internal/auth/google, so google_test.go had to use package google_test to call server.UnmarshalPrimitiveConfig. export_test.go simply exposes the unexported client field to the test package so the existing tests can compile.

If you prefer not to have export_test.go, we would need to export Client directly in google.go/generic.go. Let me know what you prefer!

File: internal/auth/google/export_test.go
Comment (camposperez-hub) [reply internal/auth/google/export_test.go:1]: export_test.go actually doesn't add any new tests—it contains zero test cases.

It's just a Go helper to avoid an import cycle: internal/server already imports internal/auth/google, so google_test.go had to use package google_test to call server.UnmarshalPrimitiveConfig. export_test.go simply exposes the unexported client field to the test package so the existing tests can compile.

If you prefer not to have export_test.go, we would need to export Client directly in google.go/generic.go. Let me know what you prefer!

File: internal/auth/generic/generic_test.go [hunk 7]
```diff
@@ -602,8 +501,13 @@ func TestValidateJwtToken(t *testing.T) {
...
 		{
-			name:        "invalid token (wrong signature)",
-			token:       "header.payload.signature",
+			name: "invalid token (wrong signature)",
+			token: generateValidToken(t, generateRSAPrivateKey(t), keyID, jwt.MapClaims{
+				"iss":   "https://example.com",
+				"aud":   "my-audience",
+				"scope": "read:files",
+				"exp":   time.Now().Add(time.Hour).Unix(),
+			}),
```
Comment (Yuan325) [inline internal/auth/generic/generic_test.go]: Should this be updated?

File: internal/auth/generic/generic_test.go [hunk 8]
```diff
@@ -659,7 +563,13 @@ func TestValidateJwtToken(t *testing.T) {
...
 			}
 			ctx := util.WithLogger(context.Background(), logger)
-			_, err = genericAuth.validateJwtToken(ctx, tc.token)
+			mcpAuth, ok := authService.(auth.MCPAuthService)
+			if !ok {
+				t.Fatalf("expected auth.MCPAuthService, got %T", authService)
+			}
+			header := make(http.Header)
+			header.Set("Authorization", "Bearer "+tc.token)
+			_, err = mcpAuth.ValidateMCPAuth(ctx, header)
```
Comment (Yuan325) [inline internal/auth/generic/generic_test.go]: Do we want to update these?

File: internal/auth/generic/generic_test.go [hunk 9]
```diff
@@ -580,11 +484,6 @@ func TestValidateJwtToken(t *testing.T) {
 		t.Fatalf("failed to initialize auth service: %v", err)
 	}
 
-	genericAuth, ok := authService.(*AuthService)
-	if !ok {
-		t.Fatalf("expected *AuthService, got %T", authService)
-	}
-
```
Comment (Yuan325) [inline internal/auth/generic/generic_test.go:587]: should we remove these?

File: internal/auth/generic/generic_test.go [hunk 10]
```diff
@@ -840,6 +750,22 @@ func TestValidateOpaqueToken(t *testing.T) {
...
+					}
+					_ = json.NewEncoder(w).Encode(config)
+					return
+				}
+				if r.URL.Path == "/jwks" {
+					w.Header().Set("Content-Type", "application/json")
+					_ = json.NewEncoder(w).Encode(map[string]any{
+						"keys": []any{},
+					})
+					return
```
Comment (Yuan325) [inline internal/auth/generic/generic_test.go]: are these relevant to the scope of migration unit tests?

File: internal/auth/generic/generic_test.go [hunk 11]
```diff
@@ -851,14 +777,16 @@ func TestValidateOpaqueToken(t *testing.T) {
...
+			cfg := generic.Config{
+				Audience:            tc.audience,
+				AuthorizationServer: server.URL,
+				ScopesRequired:      tc.scopesRequired,
+				McpEnabled:          true,
+			}
+
+			authService, err := cfg.Initialize()
+			if err != nil {
+				t.Fatalf("failed to initialize auth service: %v", err)
```
Comment (Yuan325) [inline internal/auth/generic/generic_test.go]: can we keep the original code but just update `Config: generic.Config{`

File: internal/auth/google/google_test.go [hunk 12]
```diff
@@ -199,26 +97,25 @@ func TestValidateMCPAuth_Opaque_Fallback(t *testing.T) {
...
-					}
-					respBody := fmt.Sprintf(`{"aud": %q, "azp": %q, "scope": "openid email"}`, tc.tokenInfoAud, tc.tokenInfoAzp)
-					return &http.Response{
-						StatusCode: http.StatusOK,
-						Body:       io.NopCloser(strings.NewReader(respBody)),
-						Header:     make(http.Header),
-					}, nil
-				}),
-			}
+			oldTransport := http.DefaultTransport
```
Comment (Yuan325) [inline internal/auth/google/google_test.go]: we probably dont need to update these?

**Review by Yuan325 — CHANGES_REQUESTED** [review]

File: internal/auth/generic/generic_test.go (hunk 7 above)
Comment (camposperez-hub) [reply internal/auth/generic/generic_test.go]: Yes, since validateJwtToken() is unexported, this test now runs through the public ValidateMCPAuth() method. ValidateMCPAuth() checks for a valid JWT structure; using a dummy string caused it to route to opaque token introspection instead.

File: internal/auth/generic/generic_test.go (hunk 8 above)
Comment (camposperez-hub) [reply internal/auth/generic/generic_test.go]: Yes , in package generic_test, genericAuth.validateJwtToken() is unexported and cannot be called directly. Calling ValidateMCPAuth() with an Authorization header is the public API way to test token validation without needing export_test.go.

File: internal/auth/generic/generic_test.go (hunk 9 above)
Comment (camposperez-hub) [reply internal/auth/generic/generic_test.go:587]: That assertion was only needed to call the unexported genericAuth.validateJwtToken() method. Since we now validate through the public auth.MCPAuthService interface, casting to the internal struct is no longer necessary.

File: internal/auth/generic/generic_test.go (hunk 10 above)
Comment (camposperez-hub) [reply internal/auth/generic/generic_test.go]: Because client and issuer are unexported in AuthService, we initialize via cfg.Initialize() instead of constructing the struct directly. cfg.Initialize() automatically performs OIDC discovery, so these mock endpoints allow Initialize() to succeed in package generic_test.

File: internal/auth/generic/generic_test.go (hunk 11 above)
Comment (camposperez-hub) [reply internal/auth/generic/generic_test.go]: I tried that, but AuthService contains unexported fields (client and issuer). 
In package generic_test, Go fails to compile with cannot refer to unexported field client. Using cfg.Initialize() is the standard way to instantiate AuthService from outside the package.

File: internal/auth/google/google_test.go (hunk 12 above)
Comment (camposperez-hub) [reply internal/auth/google/google_test.go]: In package google_test, a.client is also unexported and cannot be assigned directly. Mocking http.DefaultTransport lets us inject the mock response into google.AuthService cleanly without needing export_test.go.

**Review by Yuan325 — APPROVED** [review]
Thank you!
